### PR TITLE
SONAR-7855 Index active rules when creating profile from a configuration file

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileService.java [hunk 1]
```diff
@@ -40,41 +40,18 @@
   private final RuleActivator ruleActivator;
   private final QProfileFactory factory;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileService.java]: MAJOR Remove this unused "factory" private field. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1068)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [QProfileService.java#L23](https://github.com/SonarSource/sonarqube/blob/f62cb9bccf91c8e34a89f3d2433dbe61cf6a65f1/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileService.java#L23): Remove this unused import 'java.util.Map'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)
1. MINOR [QProfileService.java#L30](https://github.com/SonarSource/sonarqube/blob/f62cb9bccf91c8e34a89f3d2433dbe61cf6a65f1/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileService.java#L30): Remove this unused import 'org.sonar.db.qualityprofile.QualityProfileDto'. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. INFO [QProfileService.java#L106](https://github.com/SonarSource/sonarqube/blob/2dba8b4b16af14d4c6c0382047236ae068b0f54c/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileService.java#L106): Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)
1. INFO [QProfileService.java#L121](https://github.com/SonarSource/sonarqube/blob/2dba8b4b16af14d4c6c0382047236ae068b0f54c/server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileService.java#L121): Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java [hunk 2]
```diff
@@ -19,49 +19,164 @@
...
+import static org.sonar.server.language.LanguageTesting.newLanguages;
 
 public class CreateActionTest {
 
+  static final String XOO_LANGUAGE = "xoo";
+
+  static final RuleDto RULE = RuleTesting.newXooX1().setSeverity("MINOR").setLanguage(XOO_LANGUAGE);
+
+  @Rule
+  public DbTester dbTester = DbTester.create(System2.INSTANCE);
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java]: remark: you rename `db` -> `dbTester`. I rename `dbTester` -> `db`

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java [hunk 3]
```diff
@@ -19,49 +19,164 @@
...
+      .setParam("language", XOO_LANGUAGE)
       .setParam("name", "Yeehaw!")
       .execute();
+
     JsonAssert.assertJson(response.getInput()).isSimilarTo(getClass().getResource("CreateActionTest/create-no-importer.json"));
     assertThat(response.getMediaType()).isEqualTo(MediaTypes.JSON);
   }
+
+  @Test
+  public void create_profile_from_xml() {
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java]: great test!

File: server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java [hunk 4]
```diff
@@ -117,43 +168,33 @@ public void return_created_profile_in_json() throws Exception {
       .setParam("name", "Yeehaw!")
       .execute();
 
-    JsonAssert.assertJson(response.getInput()).isSimilarTo(getClass().getResource("CreateActionTest/create-no-importer.json"));
+    JsonAssert.assertJson(response.getInput()).isSimilarTo(getClass().getResource("CreateActionTest/test_json.json"));
```
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java]: wouldn't it be better to test the json example?

Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java]: I was sure you'll ask the question : the issue is that the WS is returning the profile key, which is generated randomly.

Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/CreateActionTest.java]: outside the scope of the PR: it seems a sign of something that needs to be extracted (and can be easily mocked)

**Comment by teryk** [comment]
LGTM

**Pull request merged**